libsweep/config.mk

```diff
@@ -1,15 +1,81 @@
 VERSION_MAJOR = 0
 VERSION_MINOR = 1
 
-UNAME := $(shell uname)
+# project directory containing src files
+SRC_DIR := src
+# project directory containing include header files
+INC_DIR := inc
+# project directory containing sweep include header files
+SWEEP_INC_DIR := inc/sweep
 
+# created directory where object files will be placed
+OBJ_DIR := obj
+# created directory where outputs will be placed
+BIN_DIR := bin
+
+# Define some commands (these might need to change depending on platform)
+MKDIR_P := mkdir -p
+
+# read the operating system
+UNAME := $(shell uname -s)
+
+# Note if the uname indicates any form of MINGW32 on windows
+ifeq ($(findstring MINGW,$(UNAME)),MINGW)
+  #if UNAME contains MINGW32
+  UNAME = MINGW
+else ifeq ($(findstring MSYS,$(UNAME)),MSYS)
+  #if UNAME contains MSYS
+  UNAME = MSYS
+endif
+
+# Set Options & Flags by Operating System
 ifeq ($(UNAME), Linux)
-  PREFIX ?= /usr
+  # For linux platforms
+  target = libsweep.so
+  dummy_target = dummy_linux
+  SRC_ARCH_DIR := src/arch/unix
+  INSTALL_DIR_LIB ?= /usr/lib
+  INSTALL_DIR_INCLUDES ?= /usr/include/sweep
+  LINKER = g++
```

Comment:
I ended up using g++ because it handled the `-mno-ms-bitfields` flag. Whereas ld threw an error:

> C:\msys32\mingw64\bin\ld.exe: unrecognised emulation mode: no-ms-bitfields

I'm not sure what the proper fix is here, but using g++ was working fine. I must have copied it over to the linux portion, I'm sure that could use ld without issue.